### PR TITLE
Add AppBar to Studio

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes** 
None (feature flagged)

**Description**
Support global app bar under feature flag

Fixes FG-1065

### CODE REVIEW

**Comment by 2metres** [comment]
Things to sort out: 

- [ ] Log out is not working any more
- [ ] Write Storybook tests for AppBar
- [ ] Figure out what to do with in app help

File: packages/studio-base/src/components/ExperimentalFeatureSettings.tsx [hunk 1]
```diff
@@ -60,6 +60,13 @@ const features: Feature[] = [
     name: "Plot panel series in settings",
     description: <>Allow editing plot panel data series in the sidebar.</>,
   },
+  {
+    key: AppSetting.ENABLE_NEW_UI,
+    name: "Enable new user interface",
+    description: (
+      <>Try out redesigned navigation and layout. Warning: We’re still figuring it our</>
+    ),
```
Comment (amacneil) [inline packages/studio-base/src/components/ExperimentalFeatureSettings.tsx:68]: Let's be a bit more specific
```suggestion
    key: AppSetting.ENABLE_NEW_TOPNAV,
    name: "Enable new top navigation",
    description: (
      <>Test our redesigned navigation bar.</>
    ),
```

**Comment by amacneil** [comment]
Help icon can be removed from the leftnav when topnav is enabled - the new help menu is sufficient.

**Comment by jtbandes** [comment]
> User-Facing Changes
> Redesign app to support global app bar

Remember that the "user-facing changes" section is meant to go in release notes. This is definitely not something we are ready to put in release notes. The user-facing changes should be none.

Should we rename "user-facing changes" to be a "release notes" section?

File: packages/studio-base/src/components/ExperimentalFeatureSettings.tsx [hunk 2]
```diff
@@ -60,6 +60,13 @@ const features: Feature[] = [
     name: "Plot panel series in settings",
     description: <>Allow editing plot panel data series in the sidebar.</>,
   },
+  {
+    key: AppSetting.ENABLE_NEW_UI,
+    name: "Enable new user interface",
+    description: (
+      <>Try out redesigned navigation and layout. Warning: We’re still figuring it our</>
```
Comment (jtbandes) [inline packages/studio-base/src/components/ExperimentalFeatureSettings.tsx:67]:
```suggestion
      <>Try out redesigned navigation and layout.</>
```

**Review by jtbandes — APPROVED** [review]

File: packages/studio-base/src/components/ExperimentalFeatureSettings.tsx (hunk 1 above)
Comment (2metres) [reply packages/studio-base/src/components/ExperimentalFeatureSettings.tsx:68]: I do think that this flag will eventually catch the sidenav removal as well. So perhap something more like "Try our redesigned navigation"

File: packages/studio-base/src/components/AppBar/Preferences.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  return (
+    <IconButton {...props} className={classes.iconButton}>
+      <SettingsOutlinedIcon />
+    </IconButton>
+  );
+}
+
+type TabOption = "general" | "privacy" | "lab";
+
+export function PreferencesDialog(props: DialogProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (2metres) [reply packages/studio-base/src/components/AppBar/Preferences.tsx:118]: This restructures the settings into groups but the underlying settings are exported from the sidebar version at the moment so should be identical

File: packages/studio-base/src/services/IAnalytics.ts [hunk 4]
```diff
@@ -21,6 +23,12 @@ enum AppEvent {
   DIALOG_CLOSE = "Studio: Dialog Closed",
   DIALOG_CLICK_CTA = "Studio: Dialog CTA Clicked",
 
+  // App Bar events
+  APPBAR_CLICK_CTA = "Studio: App Bar CTA CLicked",
+
+  // Help Menu events
+  HELPMENU_CLICK_CTA = "Studio: Help Menu CTA Clicked",
```
[the remark this answers is not in the document]
Comment (2metres) [reply packages/studio-base/src/services/IAnalytics.ts]: Not sure, just mirroring the existing ones for calls to action on the Welcome Dialog. @esthersweon what is the best way to test this?

**Comment by 2metres** [comment]
> Help icon can be removed from the leftnav when topnav is enabled - the new help menu is sufficient.
> 
> We should add the version number somewhere though - either in the new topnav help menu, or the new settings modal.

I would like to defer this one for now to a separate PR as there are still useful bits of documentation like keyboard shortcuts inside the in-app help.

File: packages/studio-base/src/components/AppBar/Preferences.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (2metres) [reply packages/studio-base/src/components/AppBar/Preferences.tsx]: Adding storybook test for this now, have a look.

File: packages/studio-base/src/services/IAnalytics.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jtbandes) [reply packages/studio-base/src/services/IAnalytics.ts]: @esthersweon feel free to change these as desired!

**Pull request merged**